### PR TITLE
Setup Backend Rails

### PR DESCRIPTION
**Pull request by bruk19**
**In this PR:-**
<hr>

- Created a new Rails API app called 'hello-rails-back-end'.
- Initialized your project with Git.
- project has a Postgres database set up.
- Created a table for storing your messages and create 5 different greetings.
- Created an API endpoint that selects a random greeting from your table

### CODE REVIEW

File: config/database.yml [hunk 1]
```diff
@@ -0,0 +1,27 @@
+# SQLite. Versions 3.8.0 and up are supported.
+#   gem install sqlite3
+#
+#   Ensure the SQLite 3 gem is defined in your Gemfile
+#   gem "sqlite3"
+#
+default: &default
+  adapter: sqlite3
```
Comment (GzimAsani) [inline config/database.yml:8]: - [ ] You need to use `Postgres` as your database

File: db/seeds.rb [hunk 2]
```diff
@@ -0,0 +1,8 @@
+# frozen_string_literal: true
+# This file should contain all the record creation needed to seed the database with its default values.
+# The data can then be loaded with the bin/rails db:seed command (or created alongside the database with db:setup).
+#
+# Examples:
+#
+#   movies = Movie.create([{ name: "Star Wars" }, { name: "Lord of the Rings" }])
```
Comment (GzimAsani) [inline db/seeds.rb:7]: - [ ] You need to create a database table for storing different greetings was created

File: app/models/application_record.rb [hunk 3]
```diff
@@ -0,0 +1,5 @@
+# frozen_string_literal: true
+
+class ApplicationRecord < ActiveRecord::Base
+  primary_abstract_class
+end
```
Comment (GzimAsani) [inline app/models/application_record.rb:5]: - [ ] You need to create a model for your greetings

File: app/controllers/api/v1/greetings_controller.rb [hunk 4]
```diff
@@ -0,0 +1,12 @@
+# frozen_string_literal: true
+
+module Api
+  module V1
+    class GreetingsController < ApplicationController
+      def index
+        greeting = Greeting.all.sample.message
+        render json: { status: 'SUCCESS', message: 'Loaded messages', greeting: }, status: :ok
```
Comment (GzimAsani) [inline app/controllers/api/v1/greetings_controller.rb]: - [ ] I am getting back an error when trying to check your greetings 
image

**Review by GzimAsani — CHANGES_REQUESTED** [review]
### Hello  @bruk19                 :wave:   
 Good job so far! There are some issues that you still need to work on to go to the next project but you are almost there!
 
 ## Highlights
 * Good job with creating a store correctly :+1:  :+1: 

 ## Required Changes 
### Frontend part:
- [ ] The router should be present on your `app.js` file where will be the `greetings` component
- [ ] Update the imports they are throwing errors
image
- [ ] The greetings are not coming on the front-end because of the problems on the backend. Hint: the greetings should be shown on the page on the front-end
- [ ] Good job displaying the link to front-end from your back-end readme file. But you should also display the link to the back-end from the front-end readme file
### Back-end
- [ ] Check the comments below :arrow_double_down: 






 ## Optional suggestions
 _Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._



Cheers and Happy coding! clapclapclap
 
 Feel free to leave any questions or comments in the PR thread if something is not 100% clear. Please, remember to tag me in your question so I can receive the notification.
 
 **_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**
 
 _As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by KayLemba — APPROVED** [review]
# STATUS: APPROVED :green_circle: :clap: 

Hello  @bruk19      , 
Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

HIGHLIGHTS:

- [x] Your project is professional 

- [x]  Your repo is professional 




 

- [x] No Linter Errors





Now you can merge this branch and move on :1st_place_medal: 
Happy Coding cheers 🥇 💯 

well-1

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**